scripts/exrrfs_upp.sh

```diff
@@ -38,45 +91,33 @@ numx=2
 
 # run the executable
-ulimit -s unlimited
-ulimit -v unlimited
-ulimit -a
-source prep_step
-${cpreq} ${EXECrrfs}/upp.x .
-${MPI_RUN_CMD} ./upp.x
+      source prep_step
+      ${MPI_RUN_CMD} ./upp.x
 # check the status copy output to COMOUT
-wrfprs="WRFPRS.GrbF${fhr}"
-wrfnat="WRFNAT.GrbF${fhr}"
-wrftwo="WRFTWO.GrbF${fhr}"
-if [[ ! -s "./${wrfprs}" ]]; then
-  echo "FATAL ERROR: failed to genereate WRF grib2 files"
-  export err=99
-  err_exit
-fi
-
-### generate final grib2 products # this part needs more work
-#WGRIB2=/apps/wgrib2/2.0.8/intel/18.0.5.274/bin/wgrib2
-#HRRR_DIR="/public/data/grib/hrrr_wrfprs/7/0/83/0_1905141_30" #COMINhrrr
-#YYJJJHH=$(date -d "${CDATE:0:8} ${CDATE:8:2}:00" +%y%j%H)
-## copy selected fields from netcdf to grib2, using HRRR grib2 as a template
-#${WGRIB2} ${HRRR_DIR}/${YYJJJHH}0000${fhr} -match ":TSOIL:" -set_grib_type simple -grib_out TSOIL_template.grib2
-#${WGRIB2} ${HRRR_DIR}/${YYJJJHH}0000${fhr} -match ":SOILW:" -set_grib_type simple -grib_out SOILW_template.grib2
-#${WGRIB2} ${HRRR_DIR}/${YYJJJHH}0000${fhr} -match ":VGTYP:" -set_grib_type simple -grib_out VGTYP.grib2
-#PY=/contrib/miniconda3/4.5.12/envs/avid_verify/bin/python
-#SCRIPTS="/lfs5/BMC/nrtrr/FIX_RRFS2/exec"
-#${PY} ${SCRIPTS}/netcdf_to_grib.py mpassit.${timestr}.nc TSLB TSOIL_template.grib2 TSOIL.grib2
-#${PY} ${SCRIPTS}/netcdf_to_grib.py mpassit.${timestr}.nc SMOIS SOILW_template.grib2 SOILW.grib2
-#cat ${wrftwo} TSOIL.grib2 SOILW.grib2 VGTYP.grib2 > ${wrftwo}.tmp
-#mv ${wrftwo}.tmp ${wrftwo}
+      fhr2=$(printf %02d ${fhr})
+      wrfprs="WRFPRS.GrbF${fhr2}"
+      wrfnat="WRFNAT.GrbF${fhr2}"
+      wrftwo="WRFTWO.GrbF${fhr2}"
+      if [[ ! -s "./${wrfprs}" ]]; then
+        echo "FATAL ERROR: failed to genereate WRF grib2 files"
+        err_exit
+      fi
 
+      mv itag itag_${fhr2}
 # Append the 2D fields onto the 3D files
-cat ${wrfprs} ${wrftwo} > ${wrfprs}.tmp
-mv ${wrfprs}.tmp ${wrfprs}
-cat ${wrfnat} ${wrftwo} > ${wrfnat}.two
-mv ${wrfnat}.two ${wrfnat}
+      cat ${wrfprs} ${wrftwo} > ${wrfprs}.tmp
+      mv ${wrfprs}.tmp ${wrfprs}
+      cat ${wrfnat} ${wrftwo} > ${wrfnat}.two
+      mv ${wrfnat}.two ${wrfnat}
 
 # copy products to COMOUT
-${cpreq} ${wrfprs} ${COMOUT}${ensindexstr}/upp/${RUN}_prs_${CDATE}_f${fhr}.grib2
-${cpreq} ${wrfnat} ${COMOUT}${ensindexstr}/upp/${RUN}_nat_${CDATE}_f${fhr}.grib2
-${cpreq} ${wrftwo} ${COMOUT}${ensindexstr}/upp/${RUN}_two_${CDATE}_f${fhr}.grib2
-${cpreq} ${wrfprs} ${COMOUT}${ensindexstr}/upp/${YYJJJHH}0000${fhr}
+      ${cpreq} ${wrfprs} ${COMOUT}${ensindexstr}/upp/${RUN}_prs_${CDATE}_f${fhr2}.grib2
+      ${cpreq} ${wrfnat} ${COMOUT}${ensindexstr}/upp/${RUN}_nat_${CDATE}_f${fhr2}.grib2
+      ${cpreq} ${wrftwo} ${COMOUT}${ensindexstr}/upp/${RUN}_two_${CDATE}_f${fhr2}.grib2
+      ln -snf  ${COMOUT}${ensindexstr}/upp/${RUN}_prs_${CDATE}_f${fhr2}.grib2 ${COMOUT}${ensindexstr}/upp/${YYJJJHH}0000${fhr2}
+
+    else
+      echo "FATAL ERROR: cannot find mpass file at ${timestr}"
+      err_exit
```

Comment:
Line 119-121: same here, we may keep the workflow running even if we miss one mpassit file.

Reply:
Right now, everything is set based on the normal runs. Mpassit should available in order. If anyone missing, that could be a problem from model. We will see if we need to add more flexibility during the test.